Use connection pool in PostgresStore

chasetopher opened the pull request:
This allows PostgresStore to be used in a threaded environment like with the puma web server.

An optional Hash with pool options can now passed when configuring the store:

```ruby
config.content_delivery = :eager_sync, :postgres, ENV['POSTGRES_CONNECTION'], {size: 5, timeout: 5}
```

The default is the same as the [connection_pool](https://github.com/mperham/connection_pool) gem, `{size: 5, timeout: 5}`.

Also, the connection pool is available through `PostgresStore#connection_pool` to allow clearing active connections before forking, useful with puma or unicorn.

I think this is a cleaner approach than #69.

gburgett commented on lib/wcc/contentful/store/postgres_store.rb:
```diff
@@ -1,53 +1,60 @@
+      @connection_pool.with do |conn|
+        Query.new(
+          self,
+          conn,
```
At this point, the `conn` object will outlive the `with` block.  Does the `with` block put the connection back into the pool?  The connection is not actually used until the query is evaluated.

chasetopher commented:
@gburgett Addressed your concerns about `Query` and added a test that uses threads, would appreciate your feedback. CI failures seem to be caused by latest Rubocop version.
